src/mongo_logger.ts

```diff
@@ -124,56 +176,247 @@ function resolveLogPath(
+  s: SeverityLevel;
+  t: Date;
+  c: MongoLoggableComponent;
+  message?: string;
+}
+
+/** @internal */
+export interface MongoDBLogWritable {
+  write(log: Log): unknown;
+}
+
+function compareSeverity(s0: SeverityLevel, s1: SeverityLevel): 1 | 0 | -1 {
+  const s0Num = SEVERITY_LEVEL_MAP.getNumericSeverityLevel(s0);
+  const s1Num = SEVERITY_LEVEL_MAP.getNumericSeverityLevel(s1);
+
+  return s0Num < s1Num ? -1 : s0Num > s1Num ? 1 : 0;
+}
+
+/** @internal */
+export interface Loggable extends Record<string, any> {
+  toLog?(): Record<string, any>;
+}
+
+function DEFAULT_LOG_TRANSFORM(logObject: Loggable): Omit<Log, 's' | 't' | 'c'> {
+  let log: Omit<Log, 's' | 't' | 'c'> = {};
+
+  const getHostPort = (address: string): { host: string; port: number } => {
+    const hostAddress = new HostAddress(address);
+
+    // NOTE: Should only default when the address is a socket address
+    const host = hostAddress.host ?? '';
+    const port = hostAddress.port ?? 0;
+    return { host, port };
+  };
+
+  const attachCommandFields = (
+    l: any,
+    ev: CommandStartedEvent | CommandSucceededEvent | CommandFailedEvent
+  ) => {
+    l.commandName = ev.commandName;
+    l.requestId = ev.requestId;
+    l.driverConnectionId = ev?.connectionId;
+    const { host, port } = getHostPort(ev.address);
+    l.serverHost = host;
+    l.serverPort = port;
+    if (ev?.serviceId) {
+      l.serviceId = ev.serviceId.toHexString();
+    }
+
+    return l;
+  };
+
+  const attachConnectionFields = (l: any, ev: ConnectionPoolMonitoringEvent) => {
+    const { host, port } = getHostPort(ev.address);
+    l.serverHost = host;
+    l.serverPort = port;
+
+    return l;
+  };
+
+  let ev;
+  switch (logObject.name) {
+    case COMMAND_STARTED:
+      ev = logObject as CommandStartedEvent;
+      log = attachCommandFields(log, ev);
+      log.message = 'Command started';
+      log.command = EJSON.stringify(ev.command);
+      log.databaseName = ev.databaseName;
+      break;
+    case COMMAND_SUCCEEDED:
+      ev = logObject as CommandSucceededEvent;
+      log = attachCommandFields(log, ev);
+      log.message = 'Command succeeded';
+      log.durationMS = ev.duration;
+      log.reply = EJSON.stringify(ev.reply);
+      break;
+    case COMMAND_FAILED:
+      ev = logObject as CommandFailedEvent;
+      log = attachCommandFields(log, ev);
+      log.message = 'Command failed';
+      log.durationMS = ev.duration;
+      log.failure = ev.failure;
+      break;
+    case CONNECTION_POOL_CREATED:
+      ev = logObject as ConnectionPoolCreatedEvent;
+      log = attachConnectionFields(log, ev);
+      log.message = 'Connection pool created';
+      if (ev.options) {
+        const { maxIdleTimeMS, minPoolSize, maxPoolSize, maxConnecting, waitQueueTimeoutMS } =
+          ev.options;
+        log = {
+          ...log,
+          maxIdleTimeMS,
+          minPoolSize,
+          maxPoolSize,
+          maxConnecting,
+          waitQueueTimeoutMS
```

Comment:
[The spec](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#pool-created-message) says these fields are only required if the user specified a value.

Are we okay with unconditionally having these fields, or do we think the extra information will be confusing if the user didn't configure it manually?

Reply:
We _could_ use the `DEFAULT_OPTIONS` map to filter some of these... I'm not sure the value is worth it, seeing defaults can be helpful too, inspect what you expect, maybe they would change across majors.